.github/workflows/tests.yml

```
@@ -22,8 +22,8 @@ jobs:
     strategy:
       fail-fast: false
       matrix:
-        os: [ubuntu-20.04]
-        pyv: ["3.8"]
+        os: [ubuntu-22.04]
+        pyv: ["3.12"]
         reqs: ["earliest", "latest", "dev"]
```

https://docs.github.com/en/actions/writing-workflows/choosing-what-your-workflow-does/running-variations-of-jobs-in-a-workflow#example-adding-configurations
```suggestion
        os: [ubuntu-22.04]
        pyv: ["3.9", "3.10", "3.11", "3.12", "3.13"]
        reqs: [earliest]
        include:
          - pyv: "3.12"
            reqs: latest
          - pyv: "3.12"
            reqs: dev
```

---

Update: in the current `earliest` requirements, many dependency versions officially support up to `py310`. I guess we should probably test all versions with `latest`, and test `py310` with `earliest` and `dev`.